### PR TITLE
[exporter/datadogexporter] Export summary count and sum as monotonic counts

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** <Describe what has changed.>

Exports summary counts and sums as monotonic counts to conform with the [Datadog Agent OpenMetrics check implementation](https://github.com/DataDog/integrations-core/blob/63ca09196d515c95deaa88bbab9fd6cf0452b801/datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/transformers/summary.py#L6). Commits can be reviewed separately.

**Is this a breaking change?**: Yes.

**Link to tracking Issue:** n/a

**Testing:** Updated unit tests; tested in end to end environment.

### CODE REVIEW

**Comment by mx-psi** [comment]
cc @KSerrania

File: exporter/datadogexporter/metrics_translator.go [hunk 1]
```diff
@@ -205,10 +207,22 @@ func mapSummaryMetrics(name string, slice pdata.SummaryDataPointSlice, quantiles
...
+			countKey := metricDimensionsToMapKey(countName, tags)
+			if dx, ok := putAndGetDiff(prevPts, countKey, ts, float64(p.Count())); ok {
+				ms = append(ms, metrics.NewCount(countName, ts, dx, tags))
+			}
+		}
+
+		{
+			sumName := fmt.Sprintf("%s.sum", name)
+			sumKey := metricDimensionsToMapKey(sumName, tags)
+			if dx, ok := putAndGetDiff(prevPts, sumKey, ts, p.Sum()); ok {
```
Comment (mx-psi) [inline exporter/datadogexporter/metrics_translator.go:222]: We assume here that sums are increasing as we do in the OpenMetrics check. See this protobuf comment for details https://github.com/open-telemetry/opentelemetry-proto/blob/8672494217bfc858e2a82a4e8c623d4a5530473a/opentelemetry/proto/metrics/v1/metrics.proto#L611-L615

**Review by KSerrania — APPROVED** [review]

**Comment by mx-psi** [comment]
This can be merged because of CODEOWNERS approval

**Comment by tigrannajaryan** [comment]
Please fix the merge conflict.

**Comment by mx-psi** [comment]
@tigrannajaryan PTAL

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**